org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/util/SemanticHighlightingTokens.xtend

```diff
@@ -0,0 +1,172 @@
+/*******************************************************************************
+ * Copyright (c) 2018 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.lsp4j.util
+
+import com.google.common.base.Preconditions
+import com.google.common.collect.ComparisonChain
+import com.google.common.collect.ImmutableList
+import java.nio.ByteBuffer
+import java.util.Base64
+import java.util.List
+import org.eclipse.xtend.lib.annotations.Data
+
+/**
+ * Utility class for encoding and decoding semantic highlighting tokens
+ * into a compact, {@code base64} representation.
+ */
+abstract class SemanticHighlightingTokens {
+
+	static val LENGTH_SHIFT = 0x0000010;
+	static val SCOPES_MASK = 0x0000FFFF;
+
+	/**
+	 * Encodes the iterable of tokens into a compact, {@code base64} string.
+	 * Returns with an empty string if the {@code tokens} argument is {@code null} or
+	 * empty.
+	 */
+	static def String encode(Iterable<? extends Token> tokens) {
```

Comment:
I have to add a encode/decode test with a huge chunk of data.

Reply:
[Done](https://github.com/eclipse/lsp4j/pull/223/files#diff-d1a1aa035a627187770faebecd6783e7R145).